gammapy/stats/fit_stat_estimator.py

```diff
@@ -0,0 +1,178 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import abc
+import numpy as np
+from scipy.optimize import brentq
+from gammapy.stats import wstat, cash
+
+__all__ = ["WStatEvaluator", "CashEvaluator"]
+
+
+class FitStatisticEvaluator(abc.ABC):
```

Comment:
I have thought again once about the name and noticed that we should maybe put the word "counts" in there, because that's how you basically explained the functionality to us in the past. We could maybe distinguish between a `FitStatistic`, which is basically just the statistic function used for (binned) likelihood fits and a `CountsStatistic`, which is what you introduce here. So what about renaming `CashEvaluator` to `CashCountsStatistic` and `WStatEvaluator` to `WStatCountsStatistic`? I think the `Evaluator` suffix might be a bit mis-leading as we already have a `MapEvaluator`, which has nothing to do with these classes here...

Reply:
Done